akka-javasdk/src/main/scala/akka/javasdk/impl/EntityDescriptorFactory.scala

```diff
@@ -23,41 +18,29 @@ import akka.javasdk.workflow.Workflow
 private[impl] object EntityDescriptorFactory extends ComponentDescriptorFactory {
 
   override def buildDescriptorFor(component: Class[_], serializer: JsonSerializer): ComponentDescriptor = {
-
-    // command handlers candidate must have 0 or 1 parameter and return the components effect type
-    // we might later revisit this, instead of single param, we can require (State, Cmd) => Effect like in Akka
-    def isCommandHandlerCandidate[E](method: Method)(implicit effectType: ClassTag[E]): Boolean = {
-      effectType.runtimeClass.isAssignableFrom(method.getReturnType) &&
-      method.getParameterTypes.length <= 1 &&
-      // Workflow will have lambdas returning Effect, we want to filter them out
-      !method.getName.startsWith("lambda$")
-    }
-
-    val commandHandlerMethods: Seq[KalixMethod] = if (classOf[EventSourcedEntity[_, _]].isAssignableFrom(component)) {
+    //TODO remove capitalization of method name, can't be done per component, because component client reuse the same logic for all
```

Comment:
Be aware of rolling updates as well. Component client of old version may be running on one node and component (entity) of new version on another node (or opposite).

Reply:
I just migrated this TODO from the existing impl so that we don't forget about this. I think a rolling update is not a big issue. Some requests will fail, ok, but the application should be ready for it anyway.